api/transport/handler.go

```diff
@@ -49,6 +51,12 @@ type HandlerSpec struct {
 	onewayHandler OnewayHandler
 }
 
+// MarshalLogObject implements zap.ObjectMarshaler.
+func (h HandlerSpec) MarshalLogObject(enc zapcore.ObjectEncoder) error {
+	enc.AddString("type", h.t.String())
+	return nil
```

Comment:
It's possible for it to fail. Consider a case where you need to fall back on reflection to encode one of the struct's fields.